lib/db/mem.js

```diff
@@ -1389,10 +1390,39 @@ module.exports = function (log, error) {
           return P.reject(error.notFound())
         }
 
+        if (! dbUtil.buffersAreEqual(recoveryKey.recoveryKeyIdHash, recoveryKeyIdHash)) {
```

Comment:
Aha, so, we actually don't need to do this constant-time-comparison stuff here.  We're comparing hashes rather than raw strings, and it doesn't matter if an attacker can harvest timing data about how close the two hashes are to matching.  We can just use the existing buffer-equality check.